docs/query-mode.md

```diff
@@ -0,0 +1,51 @@
+The Sarif.Multitool has a 'query' mode. Query mode takes a SarifLog and query expression and determines how many Results match the expression. It can write the count to the console, return the count as an exit code, signal if the count exceeds a threshold, or write an output log with the filtered results only.
```

Comment:
>mode [](start = 34, length = 4)

FWIW, the CommandLine package refers to these as verbs, and it matters a little because the terminology leaks into the help messages, for example:
```
C:\dev\sarif\WebScout\Source>sarif /help
SARIF Multitool 2.1.24
c Microsoft Corporation. All rights reserved.

ERROR(S):
  Verb '/help' is not recognized.
```

Reply:
Now "has a 'query' verb"